### ISSUE TITLE
Memory Leaks

### ISSUE THREAD

**Issue by Nomad145** [mikeboers/PyAV#61]
Greetings,

I've been noticing lots of memory leaks with PyAV when opening videos.  I've spent the good part of a day making sure I had none in my code before posting an issue, but I have a program that processes lots of video continuously using PyAV and I have noticed an increase in memory usage over the course of an hour (10GB).  I tested using the ```pyio``` branch using the default method for opening files along with the method for opening from a buffer and both resulted in the same issue.

Here are the results from one of my tests.  I'm using the ```resource``` module to return memory usage before and after two separate instances of video parsing.
```
----
Memory usage before: 12776 (kb)
No handlers could be found for logger "libav.h264"
Memory usage after: 69584 (kb)
----

----
Memory usage before: 69584 (kb)
Memory usage after: 100072 (kb)
----
```

```
 #!/usr/bin/python

import av
import resource

print "\n----"
print 'Memory usage before: %s (kb)' % resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
container = av.open('video.h264', 'r', 'h264')
for packet in container.demux():
        for frame in packet.decode():
                frame.to_image().save('/home/nomad/jpgs/frame-%02d.jpg' % frame.index)
print 'Memory usage after: %s (kb)' % resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
print "----\n"

print "\n----"
print 'Memory usage before: %s (kb)' % resource.getrusage(resource.RUSAGE_SELF).ru_maxrss

container2 = av.open('video.h264', 'r', 'h264')
for packet2 in container2.demux():
        for frame2 in packet2.decode():
                frame2.to_image().save('/home/nomad/jpgs/frame-%02d.jpg' % frame.index)

print 'Memory usage after: %s (kb)' % resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
print "----\n"
```

```
nomad@wilson ~ $ ll -h video.h264 
-rw-r--r-- 1 nomad nomad 22M Oct 14 16:26 video.h264
```

**Comment by mikeboers** [comment]
Hmm.

If you still have your test program, can you try replacing your `print frame` with `print frame.to_rgb()`, and tell us if that re-inflates it?

**Comment by mikeboers** [comment]
I'm going to assume this was fixed with #128.
